pkg/broker/statestorage/statestorage.go

```diff
@@ -55,12 +55,11 @@ func client(baseURL string, k credentials.APIKey) (*mongodbatlas.Client, error)
 	return mongodbatlas.New(hc, mongodbatlas.SetBaseURL(baseURL))
 }
 
-func Get(key credentials.APIKey, atlasURL string, realmURL string, logger *zap.SugaredLogger) (*RealmStateStorage, error) {
+func Get(ctx context.Context, key credentials.APIKey, atlasURL string, realmURL string, logger *zap.SugaredLogger) (*RealmStateStorage, error) {
 	realmClient, err := mongodbrealm.New(
-		context.TODO(),
 		nil,
 		mongodbrealm.SetBaseURL(realmURL),
-		mongodbrealm.SetAPIAuth(context.TODO(), key.PublicKey, key.PrivateKey),
+		mongodbrealm.SetAPIAuth(ctx, key.PublicKey, key.PrivateKey),
```

Comment:
correct me if I'm wrong but the `context` parameter which is passed everywhere is not used? `mongodbrealm.NewRequest` doesn't use the parameter passed. Is it planned for future?

Reply:
I'm adding context everywhere (except `mongodbrealm.New` above) right now - not sure if I should do it in this PR or a separate one, as it's a big change and not necessarily related to this error

Reply:
Ok! 👍